### DISCUSSION TITLE
Your shutdown cmd didn't finish in 2 minutes. It will be killed - I need more then 2 Minutes --> Option to set individual timer

### DISCUSSION THREAD

**Discussion by KingJim1984** [seriousm4x/UpSnap#566] [Feature requests]
Hello, I us UpSnap for my QNAP NAS. And the timer off 2 minutes to wait for it, is to short. It would be great to have the option for setting an individual timer
![grafik](https://github.com/seriousm4x/UpSnap/assets/121956934/c08bc160-d0e1-4575-bbe1-098e3ee89fbc)

**Comment by seriousm4x** [comment]
May I ask what command you are using?
Killing the command after 2 minutes is done to prevent zombie processes that will never finish. A normal wake command normally doesn't take longer than a second.

**Reply by KingJim1984** [reply]
Hello I use this command from Home Assistent to the API from UpSnap

```
shell_command:
  poweron_qnap: 'curl -X GET -H "Authorization: Bearer [TOKEN]" "http://192.168.120.6:8090/api/upsnap/wake/[id]"'
  poweroff_qnap: 'curl -X GET -H "Authorization: Bearer [TOKEN]" "http://192.168.120.6:8090/api/upsnap/shutdown/[id]"'
```
  
I trigger the command (from HomeKit over HomeAssistent):

![grafik](https://github.com/seriousm4x/UpSnap/assets/121956934/83f21f70-a417-431a-8518-4f35a6d2785d)

![grafik](https://github.com/seriousm4x/UpSnap/assets/121956934/8be65a5e-fe9d-401e-99d1-00eca9f263a7)
 
And finaly this message

![grafik](https://github.com/seriousm4x/UpSnap/assets/121956934/27623c52-75fb-4567-8fae-a04325cc6f9d)
  
I think the NAS/QNAP need more then 2 min too shutdown/start
